src/itype.h

```diff
@@ -572,7 +568,9 @@ struct itype {
     // What we're made of (material names). .size() == made of nothing.
     // MATERIALS WORK IN PROGRESS.
     std::vector<material_id> materials;
-    std::vector<use_function> use_methods; // Special effects of use
+
+    /** Actions an instance can perform (if any) indexed by action type */
+    std::map<std::string, use_function> use_methods;
```

Comment:
It may help to use a `string_id` instead of a plain string. At least for documentation purpose, showing clearly where a value is supposed to be a key into this map. But that would probably require some major refactoring, so it's outside of this PR.

Similar: the `type` of the `iuse_actor`s could be another `string_id` for the same reasons.